corehq/apps/app_manager/tests/test_extension_case.py

```diff
@@ -227,46 +242,32 @@ def test_xform_case_block_index_valid_relationship(self):
 
 class ExtensionCasesCreateOwnerID(SimpleTestCase):
 
-    def test_advanced_xform_create_owner_id_with_without_extensions(self):
-        """ Owner id should be automatically set if there are any non-extension indices"""
+    def test_advanced_xform_autoset_owner_id(self):
+        """
+            Owner id should be automatically set if there are any non-extension indices.
+            It should never be set if there are any dynamically-specified indices.
+        """
+
+        def _test_relationships(relationships, expected):
+            advanced_open_action = AdvancedOpenCaseAction.wrap({'case_indices': [{
+                'tag': 'tag{}'.format(int(random.random() * 10000)),
+                'reference_id': 'case',
+                'relationship': r,
+            } for r in relationships]})
```

Comment:
This is also required if the suggestion above is accepted.
```suggestion
            } for i, r in enumerate(relationships)]})
```